test/document.test.js

```diff
@@ -9931,4 +9931,58 @@ describe('document', function() {
       assert.ok(doc);
     });
   });
+  it('gh9880', function(done) {
+    const testSchema = new Schema({
+      prop: String,
+      nestedProp: {
+        prop: String
+      }
+    });
+    const Test = db.model('Test', testSchema);
+
+    new Test({
+      prop: 'Test',
+      nestedProp: null
+    }).save((err, doc) => {
+      doc.id;
+      doc.nestedProp;
+
+      // let's clone this document:
+      const clone = new Test({
+        prop: 'Test 2',
+        nestedProp: doc.nestedProp
+      });
+
+      // so far, so good:
+      clone.id; // 'cloned document id'
+      clone.nestedProp; // '{}'
+
+      // let's update the document:
+      Test.updateOne({
+        _id: doc._id
+      }, {
+        nestedProp: null
+      }, (err) => {
+        err;
+        // ... and retrieve it
+        Test.findOne({
+          _id: doc._id
+        }, (err, updatedDoc) => {
+
+          // now, this is interesting:
+          err;
+          updatedDoc.id; // 'document id'
+          updatedDoc.nestedProp; // 'MongooseDocument { null }'
+
+          // now this will throw a TypeError:
+          const failing = new Test({
+            prop: 'Test 3',
+            nestedProp: updatedDoc.nestedProp
+          });
+          failing;
+        });
+        done();
```

Comment:
`done()` should be on line 9982 instead of `failing;`, the test isn't done until the `new Test({ prop: 'Test 3' })` code is done running